test/OperationTest.cpp

```diff
@@ -134,3 +138,128 @@ TEST_F(OperationTestFixture, verifyCachePreventsInProgressState) {
           ParsedAsJson(HasKeyMatching("status", Eq("not started"))),
           ParsedAsJson(HasKeyMatching("status", Eq("fully materialized")))));
 }
+// _____________________________________________________________________________
+
+class StallForeverOperation : public Operation {
+  std::vector<QueryExecutionTree*> getChildren() override { return {}; }
+  string asStringImpl([[maybe_unused]] size_t) const override {
+    return "StallForEverOperation";
+  }
+  string getDescriptor() const override {
+    return "StallForEverOperationDescriptor";
+  }
+  size_t getResultWidth() const override { return 0; }
+  size_t getCostEstimate() override { return 0; }
+  uint64_t getSizeEstimateBeforeLimit() override { return 0; }
+  float getMultiplicity([[maybe_unused]] size_t) override { return 0; }
+  bool knownEmptyResult() override { return false; }
+  vector<ColumnIndex> resultSortedOn() const override { return {}; }
+  VariableToColumnMap computeVariableToColumnMap() const override { return {}; }
+
+ public:
+  using Operation::Operation;
+  // Do-nothing operation that runs for 100ms without computing anything, but
+  // which can be cancelled.
+  ResultTable computeResult() override {
+    auto end = std::chrono::steady_clock::now() + 100ms;
+    while (std::chrono::steady_clock::now() < end) {
+      checkCancellation();
```

Comment:
This requires some cooperation of the operation system to work. 
It is still legal if this always throws (although improbable) if the other thread is never scheduled in during the 100ms.
You can make it somewhat less probable by always sleeping for `1ms` between the checks (still no theoretical guarantee though).

Reply:
How come? I would turn this into an infinite loop, but in the "failure case" this would lead to an infinitely running test, so the main idea was to give this loop a 100ms limit, so if this operation is not aborted within 100ms this will cause the test to fail. Of course there is always the possibility that the thread is suspended for more than 100ms, but on a modern processor this is highly unlikely.

Reply:
I just wanted to philosophically discuss this with someone who is able to understand the implications. 
But you are of course right, in practice this always works, and you can leave the code as is.